contactsummary/tests/phpunit/CRM/Contactsummary/Utils/ContractSummaryTestTrait.php

```diff
@@ -0,0 +1,138 @@
+  }
+
+  /**
+   * Creates absence entitlements for specific leave type
+   *
+   * @param $params
+   *
+   * @return int
+   */
+  protected function createAbsenceEntitlement($params)  {
+    $period = CRM_HRAbsence_BAO_HRAbsenceEntitlement::create($params);
+    return $period->id;
+  }
+
+  /**
+   * Creates a new Job Contract for the given contact
+   *
+   * If a startDate is given, it will also create a JobDetails instance to save
+   * the contract's start date and end date(if given)
+   *
+   * @param $contactID
+   * @param null $startDate
+   * @param null $endDate
+   * @param array $extraParams
+   *
+   * @return \CRM_HRJob_DAO_HRJobContract|NULL
+   */
+  protected function createJobContract($contactID, $startDate = null, $endDate = null, $extraParams = array()) {
+    $contract = CRM_Hrjobcontract_BAO_HRJobContract::create(['contact_id' => $contactID]);
+    if($startDate) {
+      $params = [
+        'jobcontract_id' => $contract->id,
+        'period_start_date' => CRM_Utils_Date::processDate($startDate),
+        'period_end_date' => null,
+      ];
+      if($endDate) {
+        $params['period_end_date'] = CRM_Utils_Date::processDate($endDate);
+      }
+      $params = array_merge($params, $extraParams);
+      CRM_Hrjobcontract_BAO_HRJobDetails::create($params);
+    }
+    return $contract;
+  }
+
+  /**
+   * Request Leave for a specific contact
+   *
+   * @param $absenceTypeID
+   * @param $contactID
+   * @param $from
+   * @param $to
+   * @param $absence
+   *
+   * @return int
+   */
+  protected function requestLeave($absenceTypeID, $contactID, $from, $to, $absence)  {
+    $begin = new DateTime($from);
+    $end = new DateTime($to);
+    $end = $end->modify('+1 day');
+
+    $interval = DateInterval::createFromDateString('1 day');
+    $period = new DatePeriod($begin, $interval, $end);
+
+    $activityStatus = CRM_HRAbsence_BAO_HRAbsenceType::getActivityStatus('name');
+    $statusID = CRM_Utils_Array::key('Completed', $activityStatus);
+
+    $absenceTypeID = $this->getAbsenceTypeId('absence');
+    $sickTypeID = $this->getAbsenceTypeId('sick');
+
+    $activityParam = array(
+      'activity_type_id' => $sickTypeID,
+      'activity_date_time' => date('Y-m-d H:i:s'),
+      'status_id' => $statusID,
+    );
+    $mainActivity = civicrm_api3('Activity', 'create', $activityParam);
+    $sourceActivityID = $mainActivity['id'];
+
+    foreach ( $period as $dt )  {
+      $leaveDate =  $dt->format('Y-m-d');
+      $activityParam = array(
+        'source_record_id'   => $sourceActivityID,
+        'activity_type_id'   => $absenceTypeID,
+        'activity_date_time' => $leaveDate,
+        'duration'           =>  ($absence == 'full_day') ? 480 : 240,
+        'status_id'          => $statusID,
+      );
+      civicrm_api3('Activity', 'create', $activityParam);
+    }
+  }
+
+  /**
+   * Get the ID for the requested absence type using its name
+   *
+   * @param $absenceTypeName
+   *
+   * @return int
```

Comment:
This method returns a null if no absence type is found, so this should be @return int|NULL